hera/editor/lib/WinConditionCard.tsx

```diff
@@ -140,13 +140,13 @@ export default function WinConditionCard({
                   min={MIN_AMOUNT}
                   onChange={({ target: { value } }) => {
                     const amount = parseInteger(value);
-                    setAmount(amount);
                     if (amount) {
                       const newCondition = {
                         ...condition,
                         amount,
                       };
                       if (validate(newCondition)) {
+                        setAmount(amount);
                         onChange(newCondition);
```

Comment:
This change unfortunately doesn't work since it disallows invalid values while typing, like for example pressing backspace to set the amount to "" and then typing "100". Why did you make this change?

Reply:
This has to do with the fact that when you're adjusting the amount for `OptionalObjectiveAmount`, the input value changes regardless whether we have a valid number of optional objectives specified. So I wanted to make sure the user could only increment the amount up to the number of currently added optional objectives, so when the user notices the input value doesn't increment anymore, they could add more optional objectives.

For example, let's say if the user adds `OptionalObjectiveAmount` in the map editor, and specifies the value of amount to be `1`. He then goes on to add a condition that he means to be optional, but he forgets to check `optional` checkbox and then hit the `Playtest`. The UI will give the feedback `There is a problem with this map. The win conditions are not valid. [invalid-win-conditions]` and the user thinks the problem was him not ticking off the optional checkbox and he goes on and do just that. But when he clicks `Playtest` again with everything looking valid, he'll instead get another `There is a problem with this map. The win conditions are not valid. [invalid-win-conditions]`. And the reason is `condition.amount` has not been updated when the user incremented the amount from 0 to 1, right after he added the `OptionalObjectiveAmount` in the beginning, because back then, there wasn't any optional condition added yet (since the user added it after he had added `OptionalObjectiveAmount`)

I thought it'd be safe to do it since all other 'amount' objectives only checks whether the amount specified by the user is between `MIN_AMOUNT` and `MAX_AMOUNT`. Sorry, I didn't really think about the use case where the user types in the value directly to the input as you mentioned.

Do you want me to revert this change?

Reply:
Yeah I think we should revert it for now.

Reply:
done in [91dc7a9](https://github.com/nkzw-tech/athena-crisis/pull/47/commits/91dc7a98d1e43eb3f80ed7a166b5fad47d0c173e)